pandas/tests/indexes/test_base.py

```diff
@@ -1484,6 +1484,24 @@ def test_get_level_values(self):
         result = index_with_name.get_level_values('a')
         tm.assert_index_equal(result, index_with_name)
 
+    def test_unique(self):
+        idx = pd.Index([2, 3, 2, 1], name='my_index')
+        expected = pd.Index([2, 3, 1], name='my_index')
+        for level in 0, 'my_index', None:
```

Comment:
can you integrete with the other tests of unique in indexes/common.py and/or tests/test_base.py